api/api/models_api.go

```diff
@@ -100,3 +105,157 @@ func (c *ModelsController) GetModel(r *http.Request, vars map[string]string, bod
 
 	return Ok(model)
 }
+
+// DeleteModel delete a model given a project and model ID
+func (c *ModelsController) DeleteModel(r *http.Request, vars map[string]string, body interface{}) *Response {
+	ctx := r.Context()
+
+	projectID, _ := models.ParseID(vars["project_id"])
+	modelID, _ := models.ParseID(vars["model_id"])
+
+	_, err := c.ProjectsService.GetByID(ctx, int32(projectID))
+	if err != nil {
+		return NotFound(err.Error())
+	}
+
+	model, err := c.ModelsService.FindByID(ctx, modelID)
+	if err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			return NotFound(fmt.Sprintf("Model id %s not found", modelID))
+		}
+		return InternalServerError(err.Error())
+	}
+
+	// get all model version for this model
+	versions, _, err := c.VersionsService.ListVersions(ctx, modelID, c.MonitoringConfig, service.VersionQuery{})
+	if err != nil {
+		return InternalServerError(err.Error())
+	}
+
+	var allJobInModel []*models.PredictionJob
+	var allEndpointInModel []*models.VersionEndpoint
+
+	// iterate for every version, check the active prediction job / active endpoint
+	for _, version := range versions {
+		if model.Type == "pyfunc_v2" {
```

Comment:
@ariefrahmansyah, just to confirm, could you clarify if `pyfunc_v2` can only be used for batch jobs? Or is it also possible to use them for real-time models?

Reply:
@ariefrahmansyah, could you clarify this?

Reply:
Sorry I missed this. Technically, it's possible to use `pyfunc_v2` for batch jobs and real-time models. However, in Gojek itself, no real-time models are using `pyfunc_v2`.

Reply:
> Technically, it's possible to use pyfunc_v2 for batch jobs and real-time models.

TIL 😮

Reply:
Got it. This being the case, I think we should look for:
* Prediction jobs if pyfunc_v2
* Endpoints always (including pyfunc_v2)

Reply:
does this mean, pyfunc_v2 model could be deployed?

Reply:
Yep. So we need to process endpoints for all models, and if it's a `pyfunc_v2` model, we process prediction jobs for it too.

Reply:
Based on my observation on the UI, the `pyfunc_v2` model is only for prediction jobs only and not having the endpoint function (deploy), please refer [here](https://github.com/caraml-dev/merlin/blob/main/ui/src/version/VersionListTable.js#L491-L536) for the current implementation.

I will add the endpoint check for the API, but since the frontend suggest that we are not letting users deploy `pyfunc_v2` endpoints anyway. we don't need to add the endpoint table on the delete popup confirmation for deleting `pyfunc_v2` model version, wdyt about this?

Reply:
Thanks for checking that out! Yep, I don't think we'll need to expose the fact that `pyfunc_v2` models can be deployed as an endpoint to our end-users, so we can just make the change in the API server. The same can be said also for the SDK. So I think just having this change on the API would be enough :)